// We will use lite protobuf version
option optimize_for = LITE_RUNTIME;


import "timestamp.proto";

package iviLink;

message ClimateState {

  enum Limits {
    FAN_MIN_SPEED = 0;
    FAN_MAX_SPEED = 7;
    TEMP_MIN = 16;
    TEMP_MAX = 27;
  }

  required Timestamp   timestamp =  1;

  required int32 fan_speed       =  2 [ default = 5     ];
  required int32 driver_temp     =  3 [ default = 20    ];
  required int32 pass_temp       =  4 [ default = 20    ];
  required bool  auto_climate    =  5 [ default = false ];
  required bool  dual_climate    =  6 [ default = false ];
  required bool  air_recirc      =  7 [ default = false ];
  required bool  is_ac           =  8 [ default = true  ];
  required bool  front_window    =  9 [ default = false ];
  required bool  rear_window     = 10 [ default = false ];
  required bool  is_off          = 11 [ default = false ];
  required bool  fan_mode_top    = 12 [ default = false ];
  required bool  fan_mode_middle = 13 [ default = false ];
  required bool  fan_mode_lows   = 14 [ default = false ];
}

